### PR TITLE
BVTCK-165 - BVTCK-168 Additional tests for implicit unwrapping

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/BVTCK-165
 * https://hibernate.atlassian.net/browse/BVTCK-168

### CODE REVIEW

File: tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java [hunk 1]
```diff
@@ -195,6 +198,74 @@ public void validateUnwrappedValue_returns_correct_values() {
...
+		Set<ConstraintViolation<BeanWithWrapperWithTwoTypeArgumentsAndForcedUnwrapping>> constraintViolations =
+				validator.validate( new BeanWithWrapperWithTwoTypeArgumentsAndForcedUnwrapping() );
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationOf( Min.class ).withProperty( "wrapper" )
+		);
+	}
+
+	@Test(expectedExceptions = ConstraintDeclarationException.class)
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_VALIDATIONROUTINE_VALUEEXTRACTORRESOLUTION_IMPLICITUNWRAPPING, id = "c")
+	public void validate_forced_unwrapping_having_two_type_parameters_and_only_two_maximally_specific_value_extractors_raises_exception() {
```
Comment (gunnarmorling) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java]: s/only//?

Comment (gsmet) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java]: Good catch. Fixed and force pushed an update!

File: tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java [hunk 2]
```diff
@@ -122,7 +121,9 @@ public void validate_wrapped_value_if_value_extractor_unwraps_by_default_for_gen
 	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_VALIDATIONROUTINE_VALUEEXTRACTORRESOLUTION_IMPLICITUNWRAPPING, id = "a")
 	public void validate_wrapper_if_unwrapping_disabled_per_constraint_for_generic_container() {
 		Set<ConstraintViolation<WrapperWithDisabledUnwrapping>> constraintViolations = getValidatorWithValueExtractors().validate( new WrapperWithDisabledUnwrapping() );
-		assertNumberOfViolations( constraintViolations, 0 );
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationOf( Null.class ).withProperty( "integerWrapper" )
```
Comment (gunnarmorling) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java]: Wouldn't this test make more sense if it was a constraint with an implicitly unwrapping value extractor? `@Null` would be applied to the wrapper anyways.

File: tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java [hunk 3]
```diff
@@ -195,6 +198,74 @@ public void validateUnwrappedValue_returns_correct_values() {
...
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_VALIDATIONROUTINE_VALUEEXTRACTORRESOLUTION_IMPLICITUNWRAPPING, id = "c")
+	public void validate_implicit_unwrapping_having_two_type_parameters_and_only_one_maximally_specific_value_extractor_is_ok() {
+		Validator validator = TestUtil.getConfigurationUnderTest()
+				.addValueExtractor( new UnwrapByDefaultWrapperWithTwoTypeArgumentsFirstValueExtractor() )
+				.buildValidatorFactory()
+				.getValidator();
+
+		Set<ConstraintViolation<BeanWithWrapperWithTwoTypeArguments>> constraintViolations = validator.validate( new BeanWithWrapperWithTwoTypeArguments() );
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationOf( Min.class ).withProperty( "wrapper" )
```
Comment (gunnarmorling) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java]: Shouldn't this indicate the container element node somehow?

File: tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java (hunk 2 above)
Comment (gsmet) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java]: It has an implicitly unwrapping VE. See `UnwrapByDefaultWrapperValueExtractor` that is added in `getValidatorWithValueExtractors()`.

Comment (gunnarmorling) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java]: Ah, ok. Didn't notice the behavior of `getValidatorWithValueExtractors()`,  thought it'd be just the default extractors. Cool then.

**Pull request merged**